File: scratch/compare_css.js
const fs = require('fs');

// Read the HTML style block content
const htmlContent = fs.readFileSync('index.html', 'utf8');
const styleMatch = htmlContent.match(/<style>([\s\S]*?)<\/style>/);
const htmlCss = styleMatch ? styleMatch[1] : '';

// Read the external CSS content
const extCss = fs.readFileSync('assets/css/ze-ui.css', 'utf8');

// Helper to normalize CSS by stripping whitespace and comments
function normalize(css) {
    return css
        .replace(/\/\*[\s\S]*?\*\//g, '') // remove comments
        .replace(/\s+/g, ''); // remove all whitespace
}

const normHtmlCss = normalize(htmlCss);
const normExtCss = normalize(extCss);

console.log(`Normalized HTML CSS length: ${normHtmlCss.length}`);
console.log(`Normalized External CSS length: ${normExtCss.length}`);

if (normHtmlCss === normExtCss) {
    console.log("The CSS contents are identical!");
} else {
    console.log("The CSS contents are DIFFERENT.");
    
    // Find first difference
    let firstDiff = -1;
    const minLen = Math.min(normHtmlCss.length, normExtCss.length);
    for (let i = 0; i < minLen; i++) {
        if (normHtmlCss[i] !== normExtCss[i]) {
            firstDiff = i;
            break;
        }
    }
    if (firstDiff !== -1) {
        console.log(`First difference at character ${firstDiff}`);
        console.log(`HTML: ${normHtmlCss.substring(firstDiff, firstDiff + 100)}`);
        console.log(`EXT:  ${normExtCss.substring(firstDiff, firstDiff + 100)}`);
    }
}

File: scratch/compare_css.py
import re

# Read the HTML style block content
with open('index.html', 'r', encoding='utf-8') as f:
    html_content = f.read()

style_match = re.search(r'<style>(.*?)</style>', html_content, re.DOTALL)
html_css = style_match.group(1) if style_match else ''

# Read the external CSS content
with open('assets/css/ze-ui.css', 'r', encoding='utf-8') as f:
    ext_css = f.read()

# Helper to normalize CSS by stripping whitespace and comments
def normalize(css):
    css = re.sub(r'/\*.*?\*/', '', css, flags=re.DOTALL) # remove comments
    css = re.sub(r'\s+', '', css) # remove all whitespace
    return css

norm_html_css = normalize(html_css)
norm_ext_css = normalize(ext_css)

print(f"Normalized HTML CSS length: {len(norm_html_css)}")
print(f"Normalized External CSS length: {len(norm_ext_css)}")

# Check differences
if norm_html_css == norm_ext_css:
    print("The CSS contents are identical!")
else:
    print("The CSS contents are DIFFERENT.")
